src/codeu/chat/server/Server.java

```diff
@@ -57,19 +61,30 @@
   private final Uuid id;
   private final Secret secret;
 
-  private final Model model = new Model();
-  private final View view = new View(model);
   private final Controller controller;
 
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
+  private final Model model;
+  final View view;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  public Server(final Uuid id, final Secret secret, final Relay relay, final File fp) {
 
```

Comment:
Keep the parameter you have in the `Server` constructor, that does not need to change. The problem is that file is a directory, you just need to create an actual new file within it using `File(File parent, String child)`. Parent will be the already passed in directory (`fp` from the `Server` constructor) and `child` will be any filename you choose, i.e. `chatlogs.dat`, then you will use this new file to save serialized data too, in the end the code will look something like:

```
public Server(final Uuid id, final Secret secret, final Relay relay, final File dir) {
  File fp = new File(dir, 'chatlog.dat');
```

Similarly when you read the file on server startup you will have to use `new File(dir, 'chatlog.dat')`.